types/lib/apple-pay.d.ts

```diff
@@ -26,7 +26,7 @@ export type ApplePayConfig = {
    * If provided, will override `options.total` and provide the current total price on the CheckoutPricing instance
    * when the Apple Pay flow is initiated.
    */
-  pricing?: CheckoutPricingInstance;
+  pricing?: CheckoutPricingInstance | CheckoutPricingPromise;
```

Comment:
This method technically will accept a generic 'PricingInstance' or 'PricingPromise', but our intent is to focus on `CheckoutPricing` going forward.